src/data_info/services/rename.clj

```diff
@@ -16,59 +17,97 @@
   [source-path dest-path]
   (ft/path-join dest-path (ft/basename source-path)))
 
+(defn- async-thread
```

Comment:
This function does most of the work. Here's an outline:

* it's passed in an async task ID, since that's created synchronously
* it establishes an anonymous function that adds a "running" status as the move progresses
* it's passed a function that does the actual work, taking in an irods connection, the async task definition (which has the source and destination in it), and the update function
* it calls that, catching errors and updating the status as appropriate
* the completed or failed status marks the async task as complete

The update to clj-jargon allows passing an update function like the above to `move` and `move-all` (taking a path or path descriptor and an action that was performed).